pages/devs/reputers/reputers.mdx

```diff
@@ -1,150 +1,214 @@
+              "Token": "ethereum"
+          }
+      }
+    ],
 ```
-docker compose up
-```
+</Callout>
 
-If you're using Apple Silicon hardware, please make sure to use run docker-compose.arm64.yml instead.
+## Create the Truth Server
 
-### Components of the Example Reputer Node
+### Prepare the API Gateway
 
-This example contains a service for sourcing truth -- called the "`truth` service" -- a Blockless worker node for responding to Blockless requests and calculating topic-specific loss functions, and a Blockless head node for initiating requests.
+Ensure you have an API gateway or server that can accept API requests to call your model.
 
-In practice, you would only run a head node as a side car to a validator node. This way, chain validators can initiate requests to collect inferences and forecasts from workers, and ground-truth-informed loss calculations from reputers.
+### Server Responsibilities
 
-You can read more about the lifecycle of Blockless requests [on the Blockless docs](https://blockless.network/docs/protocol/networking).
+- Accept API requests from `main.py`, your custom Python script executed by the node function.
+- Respond with the corresponding inference obtained from the model.
 
-If you were just acting as a reputer, you would only be responding to requests from another peer's head node, not your own head node.
+### Truth Relay
 
+Below is a sample structure of what your `main.go`, `main.py` and Dockerfile will look like. You can also find a working example [here](https://github.com/allora-network/basic-coin-prediction-node).
 
-## Setup a Reputer Node Manually
+#### `main.go`
 
-Here, we'll recreate the steps contained within the `docker-compose.yml` file. We'll setup a reputer that sources ground truth and calculates losses for ETH price prediction topics, plus a head node for development and testing purposes.
+`allora-offchain-node` comes preconfigured with a `main.go` file inside the [`adapter/api-worker-reputer` folder](https://github.com/allora-network/allora-offchain-node/blob/dev/adapter/api-worker-reputer/main.go).
 
-First, be sure to clone and `cd` into [our example reputer repo](https://github.com/allora-network/coin-prediction-reputer).
+The `main.go` file fetches the responses outputted from the Source of Truth Endpoint based on the `SourceOfTruthEndpoint` and `Token` provided in the section above.
 
-### Generate keys
+#### `main.py`
 
-Create a set of keys for your head and worker nodes. These keys will be used in the configuration of the head and worker nodes.
+`allora-offchain-node` comes preconfigured with a `main.py` file that uses Flask to expose the Source of Truth Endpoint. 
 
-#### Create head keys
+The Flask application calls the `get_truth` function upon API request, with the required arguments (`Token`, `blockHeight`). Before proceeding, ensure that all necessary packages are listed in the `requirements.txt` file.
```

Comment:
"The Flask application serves the request from `main.go`, which is routed to the `get_inference` function, using the required arguments..."